src/server/snap.rs

```diff
@@ -451,12 +466,44 @@ impl<R: RaftExtension + 'static> Runnable for Runner<R> {
                 };
                 self.pool.spawn(task);
             }
-            Task::RecvTablet { sink, .. } => {
-                let status = RpcStatus::with_message(
-                    RpcStatusCode::UNIMPLEMENTED,
-                    "tablet snap is not supported".to_string(),
-                );
-                self.pool.spawn(sink.fail(status).map(|_| ()));
+            Task::RecvTablet { stream, sink } => {
+                if !self.can_receive_tablet_snapshot {
+                    let status = RpcStatus::with_message(
+                        RpcStatusCode::UNIMPLEMENTED,
+                        "tablet snap is not supported".to_string(),
+                    );
+                    self.pool.spawn(sink.fail(status).map(|_| ()));
+                    return;
+                }
+
+                if let Some(status) = self.receiving_busy() {
+                    self.pool.spawn(sink.fail(status));
```

Comment:
why is it difference between the code line 475?

Reply:
The original code is different. I don't know why.